features/avatar-uploader/avatar-uploader.tsx

```diff
@@ -107,9 +107,20 @@ export const AvatarUploader: FC<AvatarUploaderProps> = ({
   };
 
   const previewOrAvatar = state.preview ?? avatar;
+  const renderButtons = () => {
+    return !!previewOrAvatar;
+  };
 
   return (
-    <DEPRECATED_Modal onClose={handleClose} open={open} showCloseButton title={t.title}>
+    <ConfirmModal
+      headerTitle={t.title}
+      onCancel={handleClose}
+      onConfirm={saveAvatar}
+      onOpenChange={handleClose}
+      open={open}
+      renderCancelButton={renderButtons}
```

Comment:
Render props assume that the passed callback will return a `ReactNode`, and it's usually a rendered `JSX`. If you want to disable buttons, it is more correct to explicitly return `null` from the function instead of `boolean`. But it doesn't make sense here, since the buttons responsible for interaction with the modal window are rendered below. And they should be placed in the callback passed to the corresponding render props.